torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -120,6 +120,25 @@ class TensorPipeAgent : public RpcAgent {
 
   mutable std::mutex mutex_;
   uint64_t nextMessageID_{0};
+
+  // Struct for capturing Time-Series Metrics
+  struct TimeSeriesMetricsTracker {
+    uint64_t currentSum_;
+    uint64_t currentCount_;
+
+    explicit TimeSeriesMetricsTracker(
+        uint64_t currentSum = 0,
+        uint64_t currentCount = 0);
+
+    void addData(uint64_t dataPoint);
```

Comment:
What is a `dataPoint`?

Reply:
A datapoint is basically a new observation for the metric we are tracking (variable naming borrowed from equivalent struct in PGA here: https://github.com/pytorch/pytorch/blob/a3639fa5167e1ce88afe2f775c3fa95e4ae646e0/torch/csrc/distributed/rpc/process_group_agent.h#L126).

Reply:
I see we have a TODO in process_group_agent,
  // TODO: this class should inherit from a MetricsTracker, and can be extended
  // to track num_sends, recvs, average size of messages, etc.
Are we going to do the same with tensorpipe agent? :)

Reply:
Yeah I noticed that TODO as well - I'm guessing this was to clean up the abstractions for tracking metrics to unify across all agents. Maybe we could keep this in-line with the other agents for now, until this is refactored elsewhere?

Reply:
Yeah we can keep as it is and look at back later